zerver/lib/actions.py

```diff
@@ -3310,6 +3310,8 @@ def do_update_pointer(user_profile: UserProfile, pointer: int, update_flags: boo
                                    message__id__lte=pointer,
                                    flags=~UserMessage.flags.read)        \
                            .update(flags=F('flags').bitor(UserMessage.flags.read))
+        do_increment_logging_stat(user_profile, COUNT_STATS['messages_read_log:client:day'],
+                                  None, timezone_now(), increment=(pointer-prev_pointer))
```

Comment:
This is wrong; some messages crossed could have been read before.  You want to get the count from the previous line.